src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractImportCompletionProvider.cs

```diff
@@ -47,6 +48,9 @@ public override async Task ProvideCompletionsAsync(CompletionContext completionC
             var syntaxContext = await CreateContextAsync(document, completionContext.Position, cancellationToken).ConfigureAwait(false);
             if (!ShouldProvideCompletion(completionContext, syntaxContext))
             {
+                // Trigger a backgound task to warm up cache and return immediately.
+                // This won't cause multiple cache update to run simultaneously because WarmUpCacheAsync is implemented by AsyncBatchingWorkQueue.
+                _ = Task.Run(() => WarmUpCacheAsync(document));
```

Comment:
You shouldn't need the Task.Run.  You shoudl just add an item to the work queue.